src/rt/gtfsrt_update.cc

```diff
@@ -71,10 +85,12 @@ delay_propagation update_delay(timetable const& tt,
                                duration_t const delay,
                                std::optional<unixtime_t> const min) {
   auto const static_time = tt.event_time(r.t_, stop_idx, ev_type);
-  rtt.update_time(r.rt_, stop_idx, ev_type,
-                  min.has_value() ? std::max(*min, static_time + delay)
-                                  : static_time + delay);
-  rtt.dispatch_event_change(r.t_, stop_idx, ev_type, delay, false);
+  auto const lower_bounded_new_time = min.has_value()
+                                          ? std::max(*min, static_time + delay)
+                                          : static_time + delay;
+  rtt.update_time(r.rt_, stop_idx, ev_type, lower_bounded_new_time);
+  dispatch_event_change(rtt, r, stop_idx, ev_type, location_idx_t{0}, -1,
```

Comment:
`location_idx_t::invalid()` could be used to signal that this is a dummy value.

The sentinel value `-1` to signal no change regarding `in_out_allowed` is only specified in `abi.h` which is not included by this file. For a future reader it is not immediately clear what `-1` means. Maybe we could use an `enum enter_exit_update` or something like that to make it more readable what the value means that is passed here.